### PR TITLE
Going advanced with the bot

### PR DESCRIPTION
**Pull request by Randium**
These are the most important changes made in this pull request. This may not be the biggest pull request, but it may be the one that has the biggest amount of contributors to it. Let's keep this up, everyone!

**Most important additions of the roles branch:**
- The bot sends an error message if the user is unauthorized to use a given command;
- The ancient role check bug has (finally!) been fixed;
- Mailboxes now support freezing/abducting/suspending/muting players;
- The mechanics to start the game, they day and the night have been constructed;
- A Timer Bot is here! It will let us know when to act up.

....hang on. Are my eyes deceiving me? Did we get EXACTLY 400 additions and 100 deletions? 😱

### CODE REVIEW

**Comment by Randium** [comment]
Awwww, nooo, I screwed it up with that last commit. Quick, somebody put a useless comment somewhere. 😛

File: interpretation/ww_head.py [hunk 1]
```diff
@@ -32,6 +33,38 @@ def process(message, isGameMaster=False, isAdmin=False):
...
+            # Initiate the second round of starting the day.
+            pass
+
+        if is_command(message,['pight',False,unip]):
+            # Initiate the first round of starting the night.
+            return [Mailbox(True)]
+
+        if is_command(message,['night'],False,unip):
+            # Initiate the second round of starting the night.
+            pass
```
Comment (Bentechy66) [inline interpretation/ww_head.py]: are you supposed to `pass` here, or do you want to return a mailbox?

File: interpretation/ww_head.py [hunk 2]
```diff
@@ -795,6 +833,7 @@ def process(message, isGameMaster=False, isAdmin=False):
             '''submit'''
             # The ice king's command to submit the list of people of whom they have guessed their roles.
             if is_command(message, ['guess_that', 'freeze_all', 'submit']) and user_role == "Ice King":
+                # Note : func.freeze_all() returns a list of mailboxes. That's why there are no [] around this function.
                 return [func.freeze_all(user_id)]
```
Comment (Bentechy66) [inline interpretation/ww_head.py]: there _are_ [] around this function though

File: interpretation/ww_head.py [hunk 3]
```diff
@@ -845,6 +884,32 @@ def process(message, isGameMaster=False, isAdmin=False):
                 return todo()
             if user_role == "The Thing" and user_undead == 0:
         	    help_msg += "`" + prefix + "create_swamp` - Create swamp with chosen players. (The Thing only)\n"
+    else:
+        # I'd just like to say that I'm having to go through 600 FUCKING LINES OF CODE
+        # to get this list, purely because Randium CAN'T WRITE A FUCKNG FRAMEWORK
+        # WHY CAN'T YOU JUST USE ext.commands FROM discord.py WHICH JUST WORKS
+        # WHY DO YOU HAVE TO DO THIS FOR FUCKS SAKE
+        # ahem
```
Comment (Bentechy66) [inline interpretation/ww_head.py]: ahem

File: main.py [hunk 4]
```diff
@@ -43,7 +43,7 @@
 
 # Import config data
 import story_time.cc_creation as creation_messages
-from config import welcome_channel, game_master, dead_participant, frozen_participant, administrator
+from config import welcome_channel, game_master, dead_participant, frozen_participant, administrator, peasant
```
Comment (Bentechy66) [inline main.py]: what _is_ `peasant`? I see it all over the code, but can't find a reference

File: main.py [hunk 5]
```diff
@@ -80,18 +80,23 @@ def get_role(server_roles, target_id):
...
+    isPeasant = False
+    try:
+        if message.guild == gamelog_channel.guild:
+            role_table = [y.id for y in message.guild.get_member(message.author.id).roles]
+
+            if game_master in role_table:
+                isGameMaster = True
+            if administrator in role_table:
+                isAdmin = True
+            if peasant in role_table and message.author.bot == True:
```
Comment (Bentechy66) [inline main.py]: is `peasant` defined anywhere? Also, why are you checking if a bot is in the role table?!

File: interpretation/ww_head.py (hunk 1 above)
Comment (Randium) [reply interpretation/ww_head.py]: There's a lot that needs to happen to initiate the night, but its functions is simply not written yet. We could return the todo() function, but given that this command is only used by bots, I supposed it wouldn't be necessary to use a placeholder.

File: main.py (hunk 4 above)
Comment (Randium) [reply main.py]: Peasants are bots.

File: main.py (hunk 5 above)
Comment (Randium) [reply main.py]: It's defined in `config.py`. And we're not checking if there's a bot in the role table, we're checking that the user is a bot AND has the assigned bot role.
The bots can influence the bot quite heavily with those commands, so a double layer of protection wouldn't hurt.

**Pull request merged**